src/core/importType.js

```diff
@@ -29,29 +34,11 @@ export function isExternalModule(name, settings, path, context) {
   if (arguments.length < 4) {
     throw new TypeError('isExternalModule: name, settings, path, and context are all required');
```

Comment:
ahhh i see what you mean. I have no idea how i was able to add this exception check, but why i also didn't require `settings === context.settings` in the same check. Let's try adding that runtime check here.

or, alternatively, let's just remove the `settings` argument entirely, and not worry about the breaking change since this is plugin internals?

Reply:
[Done](https://github.com/import-js/eslint-plugin-import/pull/2334/commits/670b09633d5bf3cd2ecfe75e69199f156efd43bf)